src/wrappers/kafkajs.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * @fileoverview Handlers for kafkajs instrumentation
+ */
+
+const shimmer = require('shimmer');
+const {
+    tracer,
+    moduleUtils,
+    eventInterface,
+    utils,
+    consts,
+} = require('epsagon');
+const traceContext = require('../trace_context.js');
+
+const { EPSAGON_HEADER } = consts;
+
+/**
+ * acts as a middleware for `consumer.run()`
+ * @param {object} message the messages param to send
+ * @param {function} originalHandler original consumer function
+ */
+function kafkaMiddleware(message, originalHandler) {
+    let originalHandlerSyncErr;
+    try {
+        // Initialize tracer and runner.
+        tracer.restart();
+        const { slsEvent: kafkaEvent, startTime: kafkaStartTime } =
+        eventInterface.initializeEvent(
+            'kafka',
+            message.topic,
+            'consume',
+            'trigger'
+        );
+
+        const metadata = {
+            partition: message.partition,
+            offset: message.message.offset,
+            timestamp: new Date(parseInt(message.message.timestamp, 10)).toUTCString(),
+        };
+        if (message.message.headers[EPSAGON_HEADER]) {
+            metadata[EPSAGON_HEADER] = message.message.headers[EPSAGON_HEADER].toString();
+        }
+
+        // Convert headers from array to object and stringify them
+        const headers = Object.entries(message.message.headers).reduce((total, entry) => {
+            // eslint-disable-next-line no-param-reassign
+            total[entry[0]] = entry[1].toString();
+            return total;
+        }, {});
+
+        tracer.addEvent(kafkaEvent);
+        eventInterface.finalizeEvent(kafkaEvent, kafkaStartTime, null, metadata, {
+            headers,
+            body: message.message.value.toString(),
```

Comment:
are you sure this is always string / bytes?

Reply:
Yes:
`The "string" argument must be one of type string, Buffer, or ArrayBuffer`